interactive_engine/compiler/src/test/java/com/alibaba/graphscope/common/ir/ExpressionTest.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2020 Alibaba Group Holding Limited.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.graphscope.common.ir;
+
+import com.alibaba.graphscope.common.ir.tools.GraphBuilder;
+import com.alibaba.graphscope.common.ir.tools.GraphStdOperatorTable;
+import com.alibaba.graphscope.common.ir.tools.config.GraphOpt;
+import com.alibaba.graphscope.common.ir.tools.config.LabelConfig;
+import com.alibaba.graphscope.common.ir.tools.config.SourceConfig;
+
+import org.apache.calcite.rex.RexNode;
+import org.apache.calcite.sql.type.SqlTypeName;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ExpressionTest {
+    private GraphBuilder builder;
+
+    @Before
+    public void before() {
+        this.builder = SourceTest.mockGraphBuilder();
+    }
+
+    @Test
+    public void literal_test() {
+        RexNode node = builder.literal(1);
+        Assert.assertEquals(SqlTypeName.INTEGER, node.getType().getSqlTypeName());
+        node = builder.literal("s");
+        Assert.assertEquals(SqlTypeName.CHAR, node.getType().getSqlTypeName());
+        node = builder.literal(true);
+        Assert.assertEquals(SqlTypeName.BOOLEAN, node.getType().getSqlTypeName());
+        node = builder.literal(1.0);
+        Assert.assertEquals(SqlTypeName.DOUBLE, node.getType().getSqlTypeName());
+    }
+
+    @Test
+    public void variable_1_test() {
+        RexNode node = builder.source(mockSourceConfig(null)).variable((String) null);
+        Assert.assertEquals("DEFAULT", node.toString());
+    }
+
+    @Test
+    public void variable_2_test() {
+        RexNode node = builder.source(mockSourceConfig("a")).variable("a");
+        Assert.assertEquals("a", node.toString());
+    }
+
+    @Test
+    public void variable_3_test() {
+        try {
+            builder.source(mockSourceConfig(null)).variable("a");
+        } catch (IllegalArgumentException e) {
+            return;
+        }
+        Assert.fail("tag 'a' should not exist");
```

Comment:
`tag 'a' does not exist` is a better description.

Reply:
done